src/riak_kv_eleveldb_backend.erl

```diff
@@ -478,17 +493,19 @@ range_scan(FoldIndexFun, Buffer, Opts, #state{fold_opts=_FoldOpts,
             [] -> AdditionalOptions;
             _  -> [{range_filter, Filter} | AdditionalOptions]
         end,
-    StartK2 = [{Field, Val} || {Field, _Type, Val} <- StartK],
+    %% Pad with missing local keys after the timestamp - use minimum erlang term value of '0' to make
+    %% sure we start the search at the beginning.
+    StartK2 = [{Field, Val} || {Field, _Type, Val} <- StartK] ++ [{N, 0} || #param_v1{name = [N]} <- ExtraLK],
     StartK3 = riak_ql_ddl:make_key(Mod, LK, StartK2),
     StartK4 = riak_kv_ts_util:encode_typeval_key(StartK3), %% TODO: Avoid adding/removing type info
     StartKey = to_object_key(Bucket, StartK4),
-    EndK2 = [{Field, Val} || {Field, _Type, Val} <- EndK],
+    %% Pad the missing end - not perfect because there could be longer bitstrings, but will have to do for prototype
+    %% TODO: FIX END KEY
+    EndK2 = [{Field, Val} || {Field, _Type, Val} <- EndK] ++ [{N, <<16#ffffffff:64>>} || #param_v1{name = [N]} <- ExtraLK],
```

Comment:
I know it is going to cost some extra space, but we could tag the keys `{'a', Key}` and use `{'e', <<>>}` as the end key.

That way we create a type that allows for detecting the end key.